src/bfx_tools/bwa.ml

```diff
@@ -4,31 +4,47 @@ open Common
 module Remove = Workflow_utilities.Remove
 
 module Configuration = struct
+  type bwa_cfg_type = {
+    name: string;
+    gap_open_penalty: int;
+    gap_extension_penalty: int;
+    mismatch_penalty: int;
+  }
 
-  let default_gap_open_penalty = 11
-  let default_gap_extension_penalty = 4
-
-  module Common = struct
-    type t = {
-      name: string;
-      gap_open_penalty: int;
-      gap_extension_penalty: int;
-    }
-    let default = {
-      name = "default";
-      gap_open_penalty = default_gap_open_penalty;
-      gap_extension_penalty = default_gap_extension_penalty;
-    }
+  module Bwa_Config (D: sig val default: bwa_cfg_type end) = struct
+    type t = bwa_cfg_type
```

Comment:
To fix @ihodes issues with module names/qualified-names, I think you can replace this line with:

```ocaml
type t = bwa_cfg_type = {
  name: string;
  gap_open_penalty: int;
  gap_extension_penalty: int;
  mismatch_penalty: int;
}
```